File: task-1/src/main/java/ru/biluta/version5/Buffer.java
package ru.biluta.version5;

import java.util.LinkedList;
import java.util.Queue;

public class Buffer {
    private final Queue<Task> queue;
    private final int maxSize;

    public Buffer(int maxSize) {
        this.queue = new LinkedList<>();
        this.maxSize = maxSize;
    }

    public boolean addTask(Task task) {
        if (queue.size() < maxSize) {
            queue.add(task);
            return true;
        }
        return false;
    }

    public Task getNextTask() {
        return queue.poll();
    }

    public boolean isEmpty() {
        return queue.isEmpty();
    }

    public int getCurrentSize() {
        return queue.size();
    }

}

File: task2/src/main/kotlin/ru.biluta.task2/Buffer.kt
package main.kotlin.ru.biluta.task2

import java.util.*

class Buffer(
    private val maxSize: Int,
    private val queue: Queue<Task> = LinkedList()
) {

    fun addTask(task: Task): Boolean {
        if (queue.size < maxSize) {
            queue.add(task)
            return true
        }
        return false
    }

    fun getNextTask(): Task {
        return queue.poll()
    }

    fun isEmpty(): Boolean {
        return queue.isEmpty()
    }

    fun getCurrentSize(): Int {
        return queue.size
    }

}